Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BPMNEditorController.java

```diff
@@ -423,9 +423,12 @@ public void onEvent(final Event event) throws InterruptedException {
       args.put("elementId", elementId);
 
       ProcessService processService = (ProcessService) SpringUtil.getBean(PROCESS_SERVICE_BEAN);
-      ProcessSummaryType linkedProcess = processService.getLinkedProcess(process.getId(), elementId, currentUserType.getUsername());
+      ProcessSummaryType linkedProcess = processService.getLinkedProcess(process.getId(), elementId);
+      User user = mainC.getSecurityService().getUserById(currentUserType.getId());
+      boolean hasLinkedProcessAccess = (linkedProcess != null) &&
+          (mainC.getAuthorizationService().getProcessAccessTypeByUser(linkedProcess.getId(), user) != null);
       String linkProcessWindowPath = "static/bpmneditor/linkSubProcess.zul";
-      if (linkedProcess != null) {
+      if (hasLinkedProcessAccess) {
```

Comment:
Use ternary here for better readability?
String linkProcessWindowPath = hasLinkedProcessAccess ? "" : "";